src/smart.rs

```diff
@@ -38,30 +41,46 @@ where
         blocks_per_chunk: usize,
         min_block_size: u64,
         max_chunk_size: u64,
+    ) -> Self {
+        SmartAllocator::for_allocator(memory_properties, |id| CombinedAllocator::new(
+            id,
+            arena_size,
+            blocks_per_chunk,
+            min_block_size,
+            max_chunk_size,
+        ))
+    }
+}
+
+impl <B, A> SmartAllocator<B, A>
+where
+    B: Backend,
+    A: MemoryAllocator<B>,
+{
+    /// Create a new smart allocator from `MemoryProperties` given by a device, backed by a
+    /// custom allocator.
+    ///
+    /// ### Parameters:
+    ///
+    /// - `memory_properties`: memory properties describing the memory available on a device
+    /// - `new_allocator`: the function used to create an allocator for each memory type
+    pub fn for_allocator<F: FnMut(MemoryTypeId) -> A>(
```

Comment:
`with_allocator`. Or just plane `new` since there is no `new` method.

Reply:
There is a new that applies only to `SmartAllocator<B>` (compatible with the older API.